app/merginprojectmodel.cpp

```diff
@@ -90,25 +90,38 @@ int MerginProjectModel::rowCount( const QModelIndex &parent ) const
 void MerginProjectModel::resetProjects( const MerginProjectList &merginProjects )
 {
   beginResetModel();
-  mMerginProjects.clear();
+  ProjectList newProjects;
+
   for ( MerginProjectListEntry entry : merginProjects )
   {
-    std::shared_ptr<MerginProject> project = std::make_shared<MerginProject>();
-    project->projectNamespace = entry.projectNamespace;
-    project->projectName = entry.projectName;
-    project->serverUpdated = entry.serverUpdated;
-
-    // figure out info from local projects (projectDir etc)
-    LocalProjectInfo localProject = mLocalProjects.projectFromMerginName( entry.projectNamespace, entry.projectName );
-    if ( localProject.isValid() )
+    QString fullProjectName = MerginApi::getFullProjectName( entry.projectNamespace, entry.projectName );
+    std::shared_ptr<MerginProject> currentMerginProject = findProjectByFullName( fullProjectName );
+    if ( currentMerginProject && currentMerginProject->pending )
     {
-      project->projectDir = localProject.projectDir;
-      project->status = localProject.status;
-      // TODO: what else to copy?
+      newProjects << currentMerginProject;
```

Comment:
Could you please add a comment why we are doing this? I guess we don't want to loose the "pending" and "progress" information, anything else?

I am wondering if it would not be cleaner to still make new MerginProject like before, just have these two attributes (pending+progress) copied from the old project - just like how we copy stuff from mLocalProjects. This way it would be more explicit why we do it... what do you think?

Reply:
Don't have preferences. The other solutions would bring an update of server version, if it would have increased in a meanwhile..or even deleted (not be present anymore). So +1.

However, just realised the issue is more complex, so WIP